CMakeLists.txt

```diff
@@ -376,8 +376,29 @@ include (cmake/print_flags.cmake)
 install (EXPORT global DESTINATION cmake)
 
 add_subdirectory (contrib EXCLUDE_FROM_ALL)
+
+macro (add_executable target)
+    # invoke built-in add_executable
+    _add_executable (${ARGV})
+    get_target_property (type ${target} TYPE)
+    if (${type} STREQUAL EXECUTABLE)
+        set_property (GLOBAL APPEND PROPERTY CLICKHOUSE_EXECUTABLES ${target})
+    endif()
+endmacro()
+
+set_property (GLOBAL PROPERTY CLICKHOUSE_EXECUTABLES "")
 add_subdirectory (libs)
 add_subdirectory (utils)
+get_property (executables GLOBAL PROPERTY CLICKHOUSE_EXECUTABLES)
+foreach (executable ${executables})
+    target_link_libraries (${executable} PRIVATE ${MALLOC_LIBRARIES})
```

Comment:
Why not just link clickhouse_new_delete everywhere?

Reply:
because clickhouse_new_delete provide `operator::new/delete` symbol interposition in libcxx/stdlibc++, which is similar to malloc. I'd like to make this strategy consistent.

Reply:
> why not everywhere

because I assume only executables under dbms require memory tracking.